src/index.ts

```diff
@@ -116,6 +123,32 @@ class Server {
 
     app.providerStatus = {}
 
+    // feature registration
+    app.apiList = []
+
+    app.registerFeature = (type: string, id: string) => {
+      if (type === 'api' && !app.apiList.includes(id)) {
+        app.apiList.push(id)
+      }
+    }
+
+    // feature detection
+    app.getFeatures = () => {
+      const features = {
+        apis: [],
+        plugins: []
+      }
+      features.plugins = app.plugins.map((plugin: any) => {
```

Comment:
this returns all plugins, also ones not enabled, i think, that is not that useful

Reply:
What about it is also returned whether the plugin was enabled?
This would allow an app to know that a plugin was present and prompt the user to enable it.

> this returns all plugins, also ones not enabled, i think, that is not that useful

What about it is also returned whether the plugin was enabled?
This would allow an app to know that a plugin was present and prompt the user to enable it.

Reply:
Added query parameter to filter out disabled plugins.
Added enabled attribute to listing.